src/mono/browser/runtime/roots.ts

```diff
@@ -44,6 +45,7 @@ export function mono_wasm_new_root_buffer (capacity: number, name?: string): Was
  * Releasing this root will not de-allocate the root space. You still need to call .release().
  */
 export function mono_wasm_new_external_root<T extends MonoObject> (address: VoidPtr | MonoObjectRef): WasmRoot<T> {
+    if (WasmEnableThreads && runtimeHelpers.disableManagedTransition) throw new Error("External roots are not supported when threads are enabled");
```

Comment:
Why?

Reply:
Because [we aspire](https://github.com/dotnet/runtime/issues/100411) to detach UI thread from Mono VM and it's GC boundaries.
This is just few asserts to make sure we don't regress.

Reply:
External roots aren't GC boundaries. If anything, they're the only safe primitive to use on the UI thread. Is the idea that the root APIs are where we assert, instead of in the public APIs that consume roots?

Reply:
JS code should have no need (anymore) creating any GC roots or touching managed object via anything but `GCHandle`